.github/update_model_info.py

```diff
@@ -10,118 +10,124 @@
-                model_info[task][bundle] = {"version": "", "checksum": "", "source": ""}
-            bundle_path = os.path.join(task_path, bundle)
-            # need to add a ci test to ensure that bundles must contain `configs/metadata.json`
-            # with changelog inside it.
-            bundle_metadata_path = os.path.join(bundle_path, "configs/metadata.json")
-            if os.path.exists(bundle_metadata_path):
-                metadata = get_json_dict(bundle_metadata_path)
-                # get version number from metadata
-                latest_version = list(metadata["changelog"].keys())[0]
-                # compress bundle
-                bundle_zip_name = f"{bundle}_v{latest_version}.zip"
-                dst_path = os.path.join(temp_dir, bundle_zip_name)
-                compress_bundle(source_path=bundle_path, dst_path=dst_path)
-                # get actual checksum
-                actual_checksum = get_checksum(dst_path=dst_path, hash_func=hash_func)
-
-                # check consistency
-                if model_info[task][bundle]["version"] != latest_version:
-                    model_info[task][bundle]["version"] = latest_version
-                    changed_flag = True
-                if model_info[task][bundle]["checksum"] != actual_checksum:
-                    model_info[task][bundle]["checksum"] = actual_checksum
-                    changed_flag = True
-                # if version or checksum is changed, upload the new bundle
-                if changed_flag is True:
-                    new_source = upload_bundle(dst_path)
-                    model_info[task][bundle]["source"] = new_source
-
-    shutil.rmtree(temp_dir)
-    print(model_info)
-
-
-def compress_bundle(source_path: str, dst_path: str):
-
-    ziph = zipfile.ZipFile(dst_path, "w", zipfile.ZIP_DEFLATED)
-    for root_dir, _, filenames in os.walk(source_path):
-        ziph.write(root_dir)
-        for file in filenames:
-            ziph.write(os.path.join(root_dir, file))
-
-
-def get_checksum(dst_path: str, hash_func):
-    with open(dst_path, "rb") as f:
-        for chunk in iter(lambda: f.read(1024 * 1024), b""):
-            hash_func.update(chunk)
-    return hash_func.hexdigest()
-
-
-def upload_bundle(bundle_zip_file_path: str):
-    """
-    to be implemented:
-    1) upload to ngc/github release
-    2) return new link
     """
-    source = ""
-    return source
+    bundle_list = get_changed_bundle_list(changed_dirs)
+    models_path = "models"
+    model_info_file = "model_info.json"
+    if len(bundle_list) > 0:
+        for bundle in bundle_list:
+
+            # create a temporary copy of the bundle for further processing
+            temp_dir = tempfile.mkdtemp()
+            update_state, msg = update_model_info(
+                bundle_name=bundle, temp_dir=temp_dir, models_path=models_path, model_info_file=model_info_file
+            )
+            shutil.rmtree(temp_dir)
+
```

Comment:
Thanks @ericspod . I modify the code structure of the main function, and put the remp dir logic here. What do you think?

Reply:
I meant for the `rmtree` part to stay in the function so that you ensure you have cleanup, otherwise an exception before this line will cause the cleanup to be missed. The body of the method would be something like this:

```python
def do_something():
...
    try:
        # create temp dir
        ... # do work
        return whatever
    finally:
        shutil.rmtree(temp_dir)
```

Reply:
The way you have the code now will work and we'll not likely have an issue with it, the `try/finally` way ensures cleanup no matter if you have an exception or not so it's more of a guarantee. I'm fine with how it is now though.